Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.cpp

```diff
@@ -168,41 +196,60 @@ class PageLoaderClient final : public API::InjectedBundle::PageLoaderClient {
     }
 
 private:
-    static CString getDocumentLoaderURL(DocumentLoader* documentLoader)
-    {
-        ASSERT(documentLoader);
-        if (!documentLoader->unreachableURL().isEmpty())
-            return documentLoader->unreachableURL().string().utf8();
-
-        return documentLoader->url().string().utf8();
-    }
-
     void didStartProvisionalLoadForFrame(WebPage&, WebFrame& frame, RefPtr<API::Object>&) override
     {
-        if (!frame.isMainFrame())
-            return;
-        webkitWebPageSetURI(m_webPage, getDocumentLoaderURL(frame.coreFrame()->loader().provisionalDocumentLoader()));
+        auto webKitFrame = webkitFrameGet(&frame);
+        auto coreFrame = frame.coreFrame();
+        if (webKitFrame && coreFrame) {
```

Comment:
I don't think coreFrame can be nullptr, and we are using it without a check below